File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int price = INT_MAX;
        for (int i = 0; i < prices.size(); ++i) {
            if (prices[i] < price) {
                price = prices[i];
            } else if (prices[i] - price > profit) {
                profit = prices[i]-price;
            }
        }

        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        lowest = float("inf")
        highest = 0

        for i in range(len(prices)):
            if prices[i] < lowest:
                lowest = prices[i]
            elif prices[i] - lowest > highest:
                highest = prices[i] - lowest

        return highest